src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PersonService.kt

```diff
@@ -134,6 +152,32 @@ class PersonService(
     }
   }
 
+  fun processCandidateRecords(personEntities: List<PersonEntity>, person: Person): PersonEntity? {
+    val highConfidenceMatches: List<MatchResult> = matchService.findHighConfidenceMatches(personEntities, person)
+    return when {
+      highConfidenceMatches.isNotEmpty() -> handleMultipleHighConfidenceMatches(highConfidenceMatches)
+      else -> null
+    }
+  }
+
+  private fun handleMultipleHighConfidenceMatches(matches: List<MatchResult>): PersonEntity {
```

Comment:
Maybe we can find a better name for this - it handles single and multiple matches